reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/TimedScheduler.java

```diff
@@ -0,0 +1,274 @@
+	 * Note that only immediate submissions via {@link Scheduler#schedule(Runnable)} and
+	 * {@link Worker#schedule(Runnable)} are considered.
+	 */
+	static final String METER_TASKS_PENDING   = "scheduler.tasks.pending";
+
+	static final String TAG_SUBMISSION = "submission.type";
+
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted (via any of the
+	 * schedule methods on both {@link Scheduler} and {@link Worker}).
+	 * <p>
+	 * Note that there are actually 4 counters, which are tagged with {@link #TAG_SUBMISSION}:
+	 * {@link #SUBMISSION_DIRECT}, {@link #SUBMISSION_DELAYED}, {@link #SUBMISSION_PERIODICALLY_INITIAL}
+	 * and {@link #SUBMISSION_PERIODICALLY_ITERATION}. The sum of all these can thus be compared with the
+	 * {@link #METER_TASKS_COMPLETED} counter.
+	 */
+	static final String METER_SUBMITTED = "scheduler.tasks.submitted";
+
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted for immediate execution
+	 * (ie. {@link Scheduler#schedule(Runnable)} or {@link Worker#schedule(Runnable)}).
+	 */
+	static final String SUBMISSION_DIRECT = "direct";
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted with a delay
+	 * (ie. {@link Scheduler#schedule(Runnable, long, TimeUnit)}
+	 * or {@link Worker#schedule(Runnable, long, TimeUnit)}).
+	 */
+	static final String SUBMISSION_DELAYED = "delayed";
+	/**
+	 * {@link Counter} that increments when a task is initially submitted with a period
+	 * (ie. {@link Scheduler#schedulePeriodically(Runnable, long, long, TimeUnit)}
+	 * or {@link Worker#schedulePeriodically(Runnable, long, long, TimeUnit)}). This isn't
+	 * incremented on each re-run of the periodical task.
+	 */
+	static final String SUBMISSION_PERIODICALLY_INITIAL = "periodically_initial";
+	/**
+	 * {@link Counter} that increments by one each time a task is submitted with a period
+	 * (ie. {@link Scheduler#schedulePeriodically(Runnable, long, long, TimeUnit)}
+	 * or {@link Worker#schedulePeriodically(Runnable, long, long, TimeUnit)}), both initially AND
+	 * each time the task is re-run.
+	 */
+	static final String SUBMISSION_PERIODICALLY_ITERATION = "periodically_iteration";
+
+	final Scheduler delegate;
+
+	final MeterRegistry registry;
+
+	final Counter       submittedDirect;
+	final Counter       submittedDelayed;
+	final Counter       submittedPeriodicallyInitial;
+	final Counter       submittedPeriodicallyIteration;
+	final LongTaskTimer pendingTasks;
+	final LongTaskTimer activeTasks;
+	final Timer         completedTasks;
+
+
+	TimedScheduler(Scheduler delegate, MeterRegistry registry,  String metricPrefix, Iterable<Tag> tagsList) {
+		this.registry = registry;
+		this.delegate = delegate;
+		if (!metricPrefix.endsWith(".")) {
+			metricPrefix = metricPrefix + ".";
+		}
+
+		Tags tags = Tags.of(tagsList);
+
+		this.submittedDirect = registry.counter(metricPrefix + METER_SUBMITTED, tags.and(TAG_SUBMISSION, SUBMISSION_DIRECT));
```

Comment:
Consider storing `metricPrefix + METER_SUBMITTED` in a helper variable to increase readability, avoid future mistakes, and shorten the code.

Reply:
these used to all have different names, but now that they share the same name we can indeed do that 👍